src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ApplicationService.kt

```diff
@@ -101,6 +101,14 @@ class ApplicationService(
       validationErrors["$.document"] = "empty"
     }
 
+    if (submittedAt != null && isWomensApplication == null) {
+      validationErrors["$.isWomensApplication"] = "empty"
+    }
+
+    if (submittedAt != null && isPipeApplication == null) {
+      validationErrors["$.isPipeApplication"] = "empty"
+    }
```

Comment:
This will cause the update to fail when we save a partially completed application at the point when we don't know this information. We should probably make this nullable and only do the check at the point of submission

Reply:
So this is the point of submission as I understand it (setting submittedAt to a value) - seems the most RESTful way of doing it?

I did have a similar thought though around losing any updated data but as I understand it the frontend will call the endpoint after each section (where submittedAt would be null) then once again after the check your answers section (where submittedAt is given a value) so should be OK?